unused vars

```
politeiad/backend/gitbe/git.go:108:6:warning: stderrError declared but not used (unused)
politeiad/backend/gitbe/git.go:89:6:warning: stdoutError declared but not used (unused)
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,0 +1,14 @@
+language: go
+go:
+  - 1.8.x
+  - 1.9.x
+sudo: false
+install:
+  - go get -v github.com/golang/dep/cmd/dep
+  - dep ensure
+  - go install ./politeiad/... ./politeiawww/...
+  - go get -v github.com/alecthomas/gometalinter
+  - gometalinter --install
+script:
+  - export PATH=$PATH:$HOME/gopath/bin
+  - ./goclean.sh

--- a/README.md
+++ b/README.md
@@ -1,4 +1,5 @@
 # politeia
+[![Build Status](http://img.shields.io/travis/decred/politeia.svg)](https://travis-ci.org/decred/politeia)
 [![ISC License](http://img.shields.io/badge/license-ISC-blue.svg)](http://copyfree.org)
 
 Politeia is the Decred proposal system.

--- a/goclean.sh
+++ b/goclean.sh
@@ -1,0 +1,33 @@
+#!/bin/bash
+# The script does automatic checking on a Go package and its sub-packages, including:
+# 1. gofmt         (http://golang.org/cmd/gofmt/)
+# 2. go vet        (http://golang.org/cmd/vet)
+# 3. gosimple      (https://github.com/dominikh/go-simple)
+# 4. unconvert     (https://github.com/mdempsky/unconvert)
+# 5. ineffassign   (https://github.com/gordonklaus/ineffassign)
+# 6. race detector (http://blog.golang.org/race-detector)
+# 7. test coverage (http://blog.golang.org/cover)
+
+# gometaling (github.com/alecthomas/gometalinter) is used to run each each
+# static checker.
+
+set -ex
+
+# Make sure gometalinter is installed and $GOPATH/bin is in your path.
+# $ go get -v github.com/alecthomas/gometalinter"
+# $ gometalinter --install"
+if [ ! -x "$(type -p gometalinter)" ]; then
+  exit 1
+fi
+
+TESTDIRS=$(go list ./... | grep -v '/vendor/')
+# Automatic checks
+test -z "$(gometalinter --vendor --disable-all \
+--enable=gofmt \
+--enable=vet \
+--enable=gosimple \
+--enable=unconvert \
+--enable=ineffassign \
+--deadline=4m ./... 2>&1 | tee /dev/stderr)"
+
+env GORACE='halt_on_error=1' go test -race ${TESTDIRS}

--- a/politeiad/api/v1/identity/identity.go
+++ b/politeiad/api/v1/identity/identity.go
@@ -105,12 +105,7 @@ func (fi *FullIdentity) Save(filename string) error {
 	if err != nil {
 		return fmt.Errorf("could not marshal identity")
 	}
-	err = ioutil.WriteFile(filename, id, 0600)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return ioutil.WriteFile(filename, id, 0600)
 }
 
 func (fi *FullIdentity) SignMessage(message []byte) [SignatureSize]byte {
@@ -167,12 +162,7 @@ func (pi *PublicIdentity) SavePublicIdentity(filename string) error {
 	if err != nil {
 		return fmt.Errorf("could not marshal public identity")
 	}
-	err = ioutil.WriteFile(filename, id, 0600)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return ioutil.WriteFile(filename, id, 0600)
 }
 
 // Zero out a byte slice.

--- a/politeiad/backend/gitbe/database.go
+++ b/politeiad/backend/gitbe/database.go
@@ -159,12 +159,8 @@ func (g *gitBackEnd) writeAnchorRecord(key [sha256.Size]byte, anchor Anchor) err
 	// Use a batch for now
 	batch := new(leveldb.Batch)
 	batch.Put(k, la)
-	err = g.db.Write(batch, nil)
-	if err != nil {
-		return err
-	}
 
-	return nil
+	return g.db.Write(batch, nil)
 }
 
 // readAnchorRecord retrieves the anchor record based on the provided merkle
@@ -233,12 +229,8 @@ func (g *gitBackEnd) writeLastAnchorRecord(lastAnchor LastAnchor) error {
 	// Use a batch for now
 	batch := new(leveldb.Batch)
 	batch.Put([]byte(LastAnchorKey), la)
-	err = g.db.Write(batch, nil)
-	if err != nil {
-		return err
-	}
 
-	return nil
+	return g.db.Write(batch, nil)
 }
 
 // readLastAnchorRecord retrieves the last anchor record.
@@ -330,12 +322,8 @@ func (g *gitBackEnd) writeUnconfirmedAnchorRecord(unconfirmed UnconfirmedAnchor)
 	// Use a batch for now
 	batch := new(leveldb.Batch)
 	batch.Put([]byte(UnconfirmedKey), ua)
-	err = g.db.Write(batch, nil)
-	if err != nil {
-		return err
-	}
 
-	return nil
+	return g.db.Write(batch, nil)
 }
 
 // readUnconfirmedAnchorRecord retrieves the unconfirmed anchor record.

--- a/politeiad/backend/gitbe/git.go
+++ b/politeiad/backend/gitbe/git.go
@@ -62,10 +62,7 @@ func (g *gitBackEnd) git(path string, args ...string) ([]string, error) {
 	}
 
 	ge.cmd = append(ge.cmd, g.gitPath)
-	for _, v := range args {
-		ge.cmd = append(ge.cmd, v)
-	}
-
+	ge.cmd = append(ge.cmd, args...)
 	if g.gitTrace {
 		defer func() { ge.log() }()
 	}
@@ -134,6 +131,15 @@ func (g *gitBackEnd) git(path string, args ...string) ([]string, error) {
 	// have completed.
 	wg.Wait()
 
+	if stdoutError != nil {
+		ge.err = fmt.Errorf("scanner error on stdout: %v", err)
+		return nil, ge
+	}
+	if stderrError != nil {
+		ge.err = fmt.Errorf("scanner error on stderr: %v", err)
+		return nil, ge
+	}
+
 	// Finish up cmd.
 	err = cmd.Wait()
 	if err != nil {
@@ -160,38 +166,22 @@ func (g *gitBackEnd) gitVersion() (string, error) {
 
 func (g *gitBackEnd) gitAdd(path, filename string) error {
 	_, err := g.git(path, "add", filename)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return err
 }
 
 func (g *gitBackEnd) gitCommit(path, message string) error {
 	_, err := g.git(path, "commit", "-m", message)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return err
 }
 
 func (g *gitBackEnd) gitCheckout(path, branch string) error {
 	_, err := g.git(path, "checkout", branch)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return err
 }
 
 func (g *gitBackEnd) gitBranchDelete(path, branch string) error {
 	_, err := g.git(path, "branch", "-D", branch)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return err
 }
 
 func (g *gitBackEnd) gitBranches(path string) ([]string, error) {
@@ -224,11 +214,7 @@ func (g *gitBackEnd) gitPull(path string, fastForward bool) error {
 
 func (g *gitBackEnd) gitRebase(path, branch string) error {
 	_, err := g.git(path, "rebase", branch)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return err
 }
 
 func (g *gitBackEnd) gitPush(path, remote, branch string, upstream bool) error {
@@ -247,11 +233,7 @@ func (g *gitBackEnd) gitPush(path, remote, branch string, upstream bool) error {
 
 func (g *gitBackEnd) gitNewBranch(path, branch string) error {
 	_, err := g.git(path, "checkout", "-b", branch)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return err
 }
 
 func (g *gitBackEnd) gitLastDigest(path string) ([]byte, error) {
@@ -313,11 +295,7 @@ func (g *gitBackEnd) gitClone(from, to string) error {
 
 	log.Infof("Cloning git repo %v to %v", from, to)
 	_, err = g.git("", "clone", from, to)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return err
 }
 
 // gitInit initializes a new repository.  If the repository exists
@@ -371,10 +349,6 @@ func (g *gitBackEnd) gitInitRepo(path string) error {
 	if err != nil {
 		return err
 	}
-	err = g.gitCommit(path, "Add .gitignore")
-	if err != nil {
-		return err
-	}
 
-	return nil
+	return g.gitCommit(path, "Add .gitignore")
 }

--- a/politeiad/backend/gitbe/git_test.go
+++ b/politeiad/backend/gitbe/git_test.go
@@ -209,6 +209,9 @@ func TestFsck(t *testing.T) {
 	// Uncompress
 	b := bytes.NewBuffer(xxx)
 	r, err := zlib.NewReader(b)
+	if err != nil {
+		t.Fatal(err)
+	}
 	buf := new(bytes.Buffer)
 	w := bufio.NewWriter(buf)
 	io.Copy(w, r)

--- a/politeiad/backend/gitbe/gitbe.go
+++ b/politeiad/backend/gitbe/gitbe.go
@@ -322,13 +322,8 @@ func updatePSR(path, id string, psr *backend.ProposalStorageRecord) error {
 		return err
 	}
 	defer f.Close()
-	encoder := json.NewEncoder(f)
-	err = encoder.Encode(*psr)
-	if err != nil {
-		return err
-	}
 
-	return nil
+	return json.NewEncoder(f).Encode(*psr)
 }
 
 // commitPSR commits the PSR into a git repo.
@@ -344,12 +339,7 @@ func (g *gitBackEnd) commitPSR(path, id, msg string) error {
 	}
 
 	// git commit -m "message"
-	err = g.gitCommit(path, "Update proposal status "+id+" "+msg)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return g.gitCommit(path, "Update proposal status "+id+" "+msg)
 }
 
 // deltaCommits returns sha1 extended digests and one line commit messages to
@@ -612,7 +602,7 @@ func (g *gitBackEnd) anchorAllRepos() error {
 			log.Errorf("anchorAllRepos unlock error: %v", err)
 		}
 	}()
-	if g.shutdown == true {
+	if g.shutdown {
 		return fmt.Errorf("anchorAllRepos: %v", backend.ErrShutdown)
 	}
 
@@ -841,12 +831,7 @@ func (g *gitBackEnd) afterAnchorVerify(vrs []v1.VerifyDigest, precious [][]byte)
 
 	// Update database record
 	ua := UnconfirmedAnchor{Merkles: precious}
-	err = g.writeUnconfirmedAnchorRecord(ua)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return g.writeUnconfirmedAnchorRecord(ua)
 }
 
 // safeReadUnconfirmedAnchorRecord is a wrapper around
@@ -970,7 +955,7 @@ func (g *gitBackEnd) New(name string, files []backend.File) (*backend.ProposalSt
 			log.Errorf("Unlock error: %v", err)
 		}
 	}()
-	if g.shutdown == true {
+	if g.shutdown {
 		return nil, backend.ErrShutdown
 	}
 
@@ -1065,7 +1050,7 @@ func (g *gitBackEnd) getProposalLock(token []byte, repo string, includeFiles boo
 			log.Errorf("Unlock error: %v", err)
 		}
 	}()
-	if g.shutdown == true {
+	if g.shutdown {
 		return nil, backend.ErrShutdown
 	}
 
@@ -1317,7 +1302,7 @@ func (g *gitBackEnd) SetUnvettedStatus(token []byte, status backend.PSRStatusT) 
 			log.Errorf("Unlock error: %v", err)
 		}
 	}()
-	if g.shutdown == true {
+	if g.shutdown {
 		return backend.PSRStatusInvalid, backend.ErrShutdown
 	}
 
@@ -1484,7 +1469,7 @@ func (g *gitBackEnd) Inventory(vettedCount, branchCount uint, includeFiles bool)
 			log.Errorf("Unlock error: %v", err)
 		}
 	}()
-	if g.shutdown == true {
+	if g.shutdown {
 		return nil, nil, backend.ErrShutdown
 	}
 
@@ -1619,13 +1604,9 @@ func (g *gitBackEnd) newLocked() error {
 	if err != nil {
 		return err
 	}
-	log.Infof("Running dcrtime fsck on vetted repository")
-	err = g.fsck(g.vetted)
-	if err != nil {
-		return err
-	}
 
-	return nil
+	log.Infof("Running dcrtime fsck on vetted repository")
+	return g.fsck(g.vetted)
 }
 
 // New returns a gitBackEnd context.  It verifies that git is installed.

--- a/politeiad/backend/gitbe/gitbe_test.go
+++ b/politeiad/backend/gitbe/gitbe_test.go
@@ -65,6 +65,9 @@ func TestAnchorWithCommits(t *testing.T) {
 
 	// Initialize stuff we need
 	g, err := New(dir, "", "", testing.Verbose())
+	if err != nil {
+		t.Fatal(err)
+	}
 	g.test = true
 
 	// Create 5 unvetted proposals
@@ -335,7 +338,7 @@ func TestAnchorWithCommits(t *testing.T) {
 
 	// Vet + anchor
 	t.Logf("===== INTERLEAVE ANCHORS =====")
-	status, err = g.SetUnvettedStatus(psr[2].Token, backend.PSRStatusVetted)
+	_, err = g.SetUnvettedStatus(psr[2].Token, backend.PSRStatusVetted)
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -345,7 +348,7 @@ func TestAnchorWithCommits(t *testing.T) {
 	}
 
 	// Vet + anchor
-	status, err = g.SetUnvettedStatus(psr[0].Token, backend.PSRStatusVetted)
+	_, err = g.SetUnvettedStatus(psr[0].Token, backend.PSRStatusVetted)
 	if err != nil {
 		t.Fatal(err)
 	}

--- a/politeiad/cmd/politeia/politeia.go
+++ b/politeiad/cmd/politeia/politeia.go
@@ -19,8 +19,8 @@ import (
 	"strings"
 	"time"
 
-	"github.com/decred/dcrtime/merkle"
 	"github.com/decred/dcrd/dcrutil"
+	"github.com/decred/dcrtime/merkle"
 	"github.com/decred/politeia/politeiad/api/v1"
 	"github.com/decred/politeia/politeiad/api/v1/identity"
 	"github.com/decred/politeia/util"
@@ -728,7 +728,6 @@ func _main() error {
 			default:
 				return fmt.Errorf("invalid action: %v", a)
 			}
-			continue
 		}
 	}
 

--- a/politeiad/config.go
+++ b/politeiad/config.go
@@ -17,8 +17,8 @@ import (
 	"strings"
 
 	flags "github.com/btcsuite/go-flags"
-	"github.com/decred/dcrtime/api/v1"
 	"github.com/decred/dcrd/dcrutil"
+	"github.com/decred/dcrtime/api/v1"
 	"github.com/decred/politeia/util"
 )
 
@@ -494,7 +494,7 @@ func loadConfig() (*config, []string, error) {
 		if err != nil {
 			return nil, nil, err
 		}
-		cfg.RPCUser = base64.StdEncoding.EncodeToString([]byte(name))
+		cfg.RPCUser = base64.StdEncoding.EncodeToString(name)
 		log.Warnf("RPC user name not set, using random value")
 	}
 	if cfg.RPCPass == "" {
@@ -502,7 +502,7 @@ func loadConfig() (*config, []string, error) {
 		if err != nil {
 			return nil, nil, err
 		}
-		cfg.RPCPass = base64.StdEncoding.EncodeToString([]byte(pass))
+		cfg.RPCPass = base64.StdEncoding.EncodeToString(pass)
 		log.Warnf("RPC password not set, using random value")
 	}
 

--- a/politeiawww/backend.go
+++ b/politeiawww/backend.go
@@ -341,12 +341,8 @@ func (b *backend) verifyResetPassword(user *database.User, rp www.ResetPassword,
 	user.NewUserVerificationToken = nil
 	user.NewUserVerificationExpiry = 0
 	user.HashedPassword = hashedPassword
-	err = b.db.UserUpdate(*user)
-	if err != nil {
-		return err
-	}
 
-	return nil
+	return b.db.UserUpdate(*user)
 }
 
 // LoadInventory fetches the entire inventory of proposals from politeiad

